app/services/qa/iri_template_service.rb

```diff
@@ -24,9 +22,8 @@ def self.build_url(url_config:, substitutions:)
       url = url_config.template
       url_config.mapping.each do |m|
         key = m.variable
-        url = url.gsub("{?#{key}}", m.simple_value(substitutions[key])) # Incorrectly applies pattern {?var_name} to produce substitution 'value'
-        # url.gsub("{#{key}}", m.simple_value(substitutions[key]))  # TODO: pattern {var_name} should produce substitution 'value'
-        # url.gsub("{?#{key}}", m.parameter_value(substitutions[key])) # TODO: pattern {?var_name} should produce substitution 'var_name=value'
+        url = url.gsub("{#{key}}", m.simple_value(substitutions[key]))
+        url = url.gsub("{?#{key}}", m.parameter_value(substitutions[key]))
```

Comment:
Use the proper substitution based on whether the template uses `{?var}` or `{var}`